go/plugins/postgresql/docstore.go

```diff
@@ -118,40 +166,119 @@ func (ds *docStore) Retrieve(ctx context.Context, req *ai.RetrieverRequest) (*ai
 	return res, nil
 }
 
+
+type IndexerOptions struct {
+	VectorStore *VectorStore
```

Comment:
See retriever for comments around not nesting structs/methods. The Docstore will have methods Retrieve and Index, this will how the logic for querying the DB and adding docs respectively.